### PR TITLE
DuckDuckGoSearch News Tool

### PR DESCRIPTION
**Pull request by timonpalm**
Description: 
I wanted to use the DuckDuckGoSearch tool in an agent to let him get the latest news for a topic. DuckDuckGoSearch has already an implemented function for retrieving news articles. But there wasn't a tool to use it. I simply adapted the SearchResult class with an extra argument "backend". You can set it to "news" to only get news articles.

Furthermore, I added an example to the DuckDuckGo Notebook on how to further customize the results by using the DuckDuckGoSearchAPIWrapper.

Dependencies: no new dependencies

@hinthornw

### CODE REVIEW

File: libs/langchain/langchain/tools/ddg_search/tool.py [hunk 1]
```diff
@@ -56,14 +56,24 @@ class DuckDuckGoSearchResults(BaseTool):
...
     ) -> str:
         """Use the tool."""
-        return str(self.api_wrapper.results(query, self.num_results))
+        res = self.api_wrapper.results(query, self.num_results, backend=self.backend)
+
+        def dict2str(d: dict):
+            items = []
+            for k, v in d.items():
+                items.append(f"{k}: {v}")
+            return ", ".join(items)
```
Comment (leo-gan) [inline libs/langchain/langchain/tools/ddg_search/tool.py:73]: dict to str:
`str(d)`
or
`json.dums(d)`

**Review by leo-gan — APPROVED** [review]
LGTM

File: libs/langchain/langchain/tools/ddg_search/tool.py (hunk 1 above)
Comment (timonpalm) [reply libs/langchain/langchain/tools/ddg_search/tool.py:73]: Yes, had this before, but all the keys and values are in quotation marks. Though that could result in an ugly prompt when using the tool in an agent and might lead to worse results in a LLM?! Also, with this approach the output is more similar to the output of DuckDuckGoSearchRun

**Comment by baskaryan** [comment]
thanks @timonpalm!

**Pull request merged**